### PR TITLE
Improve SchedulerStatus code and test as follow-up

### PR DESCRIPTION
**Pull request by ryoqun**
(this isn't a blocking pr for enabling unified scheduler: #1668)

#### Problem

There's some nice-to-have changes after #1690

#### Summary of Changes

ship it as a bonus before or after branching v2.0.

### CODE REVIEW

File: runtime/src/installed_scheduler_pool.rs [hunk 1]
```diff
@@ -549,7 +551,8 @@ impl BankWithSchedulerInner {
                 let scheduler = self.scheduler.read().unwrap();
                 // Re-register a new timeout listener only after acquiring the read lock;
                 // Otherwise, the listener would again put scheduler into Stale before the read
-                // lock under an extremely-rare race condition, causing panic below.
+                // lock under an extremely-rare race condition, causing panic below in
+                // active_scheduler().
                 pool.register_timeout_listener(self.do_create_timeout_listener());
```
Comment (ryoqun) [inline runtime/src/installed_scheduler_pool.rs:556]: before this pr, commenting out this `register_timeout_listener()` won't fail any tests... ;)

not anymore

File: unified-scheduler-pool/src/sleepless_testing.rs [hunk 2]
```diff
@@ -61,61 +63,97 @@ mod sleepless_testing_real {
...
+                Greater => {
+                    trace!("Blocked on {} at {:?}", anchored_check_point, current());
+                    // anchor is one of future check points; block the current thread until
+                    // that happens
+                    let wait_result;
+                    (current_index, wait_result) = self
+                        .condvar
+                        .wait_timeout_while(
+                            current_index,
+                            Duration::from_secs(3),
```
Comment (apfitzge) [inline unified-scheduler-pool/src/sleepless_testing.rs]: from interface perspective 3s seems arbitrary and very odd to me.

Comment (ryoqun) [reply unified-scheduler-pool/src/sleepless_testing.rs]: hehe, the 3s is defined as the maximum duration impatient me can wait before hitting Ctrl+C on terminal.

joking aside, it's indeed odd. but the motto of sleepless_testing is faster unit testing. so, still a step forward than the status quo (indefinite blocking), imo.

that said, if changing to 10sec isn't acceptable still. I'll revert this nicety to be added later and land the others be merged in this pr...

Comment (apfitzge) [reply unified-scheduler-pool/src/sleepless_testing.rs]: Yeah imo would rather not clutter this library-esque interface; iirc you said would eventually be splitting this to a separate crate, so hesitant to add any behavior here that we would not want in a separate crate.

Comment (ryoqun) [reply unified-scheduler-pool/src/sleepless_testing.rs]: okay, reverted the timeout part: 124917000bef569ba61566743c7c6e1cb57a4b4c

**Review by apfitzge — APPROVED** [review]

**Pull request merged**